### PR TITLE
Fix scrap adjustment affect the global key rate

### PR DESCRIPTION
**Pull request by sumet123**
- Fix scrap adjustment affect the global key rate

### CODE REVIEW

File: src/classes/Commands.ts [hunk 1]
```diff
@@ -1295,7 +1295,7 @@ export = class Commands {
         }
 
         this.bot.pricelist
-            .addPrice(params as EntryData, true)
+            .addPrice(params as EntryData, true, params.sku === '5021;6' ? 'manual' : null)
```
Comment (idinium96) [inline src/classes/Commands.ts:1298]: what if they add keys with autoprice enabled?

File: src/classes/Commands.ts [hunk 2]
```diff
@@ -1653,7 +1653,7 @@ export = class Commands {
         }
 
         this.bot.pricelist
-            .updatePrice(entryData, true)
+            .updatePrice(entryData, true, entryData.sku === '5021;6' ? 'manual' : null)
```
Comment (idinium96) [inline src/classes/Commands.ts:1656]: same as above. update to enable autoprice

**Comment by idinium96** [comment]
I think it's better to replace `src: string | null = null` in the add/update functions to `isScrapAdjustment = false`, then on the `!add` and `!update` command, it will be unspecified (false), and at Autokeys things, just put `this.isEnableScrapAdjustment`, and make correct logics in validate function.

**Comment by sumet123** [comment]
> I think it's better to replace `src: string | null = null` in the add/update functions (Pricelist.ts) to `isScrapAdjustment = false`, then on the `!add` and `!update` command (Commands.ts), it will be unspecified (false), and at Autokeys.ts, just put `this.isEnableScrapAdjustment`, and make correct logics in validate function in Pricelist.ts.

My first thought is there might be other modules that use `updatePrice()` with different logics so using `isScrapAdjustment` boolean is a bit too specific. (with this idea, the `src` should not be `null` anyways) But for now, I agree that using what you purposed will be more logical.

File: src/classes/Commands.ts (hunk 1 above)
Comment (sumet123) [reply src/classes/Commands.ts:1298]: Should be fine? As it won't update key rate (check the logic in `Pricelist.ts`) but the `src` could misleading.

**Comment by sumet123** [comment]
Is the new version any better?

**Comment by idinium96** [comment]
looks like it's better. Have you tried it?

**Pull request merged**